tsl/test/sql/partialize_finalize.sql

```diff
@@ -182,8 +182,24 @@ select length(_timescaledb_internal.partialize_agg(min(a+1))) from foo;
 \set ON_ERROR_STOP 0
 select length(_timescaledb_internal.partialize_agg(1+min(a))) from foo;
 select length(_timescaledb_internal.partialize_agg(min(a)+min(a))) from foo;
+--non-trivial HAVING clause not allowed with partialize_agg
+select time_bucket('1 hour', b) as b, _timescaledb_internal.partialize_agg(avg(a))
+from foo
+group by 1
+having avg(a) > 3;
+--mixing partialized and non-partialized aggs is not allowed
+select time_bucket('1 hour', b) as b, _timescaledb_internal.partialize_agg(avg(a)), sum(a)
+from foo
+group by 1;
 \set ON_ERROR_STOP 1
 
+--partializing works with HAVING when the planner can effectively
+--reduce it. In this case to a simple filter.
+select time_bucket('1 hour', b) as b, toastval, _timescaledb_internal.partialize_agg(avg(a))
+from foo
+group by b, toastval
+having toastval LIKE 'does not exist';
```

Comment:
comment doesn't reflect the test. shouldn't this be  `HAVING  ...partialize_agg(avg(a))  > ... ` ?

Reply:
Could you add 2 more tests:
1.  select ... , partialize_agg( AGGREGATE1), partialize_agg( AGGREGATE2)
 ....
HAVING partialize_agg(AGGREGATE3) ---> all 3 aggregates are different

2. select partialize_agg(AGGREGATE1), partialize_gg(AGGREAGTE2)
...
HAVING partialize_agg(AGGREGATE2)  --->same as 1 in target list

where AGGREGATE1, AGGREGATE2, AGGREGATE3 are different partializable aggregates.

Reply:
> comment doesn't reflect the test. shouldn't this be `HAVING ...partialize_agg(avg(a)) > ... ` ?

Not sure what you mean. The point of this test is to show that HAVING quals that are filters can be accepted since the planner reduces them to filters on the base rel, thus internally removing the HAVING qual. If there are any remaining HAVING quals, the query will fail since this PR makes it so that HAVING quals aren't accepted together with partializaion.

Reply:
> Could you add 2 more tests:
> 
>     1. select ... , partialize_agg( AGGREGATE1), partialize_agg( AGGREGATE2)
>        ....
>        HAVING partialize_agg(AGGREGATE3) ---> all 3 aggregates are different
> 
>     2. select partialize_agg(AGGREGATE1), partialize_gg(AGGREAGTE2)
>        ...
>        HAVING partialize_agg(AGGREGATE2)  --->same as 1 in target list
> 
> 
> where AGGREGATE1, AGGREGATE2, AGGREGATE3 are different partializable aggregates.

FWIW, the above expression/queries are non-sensible since HAVING must be a qual (boolean expression). Thus `partialize_agg(avg(temp) > 3)` doesn't works since that returns `bytea` and not a boolean, and, `partialize_agg(avg(temp)) > 3` doesn't work either for similar reasons.

Reply:
I wanted to have a test similar to the cont. aggregate views. But see that cont. aggregates use finalize_agg (not partialize_agg) in the final query. My bad.

Reply:
No worries, I am happy to add other tests if you come up with something.